src/realm/sync/instruction_applier.cpp

```diff
@@ -702,56 +726,81 @@ void InstructionApplier::operator()(const Instruction::ArrayMove& instr)
-                                    instr.prior_size);
-            }
-            if (index >= list.size()) {
-                bad_transaction_log("ArrayErase: Index out of bounds (%1 >= %2)", index, list.size());
-            }
-            if (instr.prior_size != list.size()) {
-                bad_transaction_log("ArrayErase: Invalid prior_size (list size = %1, prior_size = %2)", list.size(),
-                                    instr.prior_size);
-            }
-
-            list.remove(index, index + 1);
-        },
-        [&](auto&&...) {
-            bad_transaction_log("Invalid path for ArrayErase");
-        },
-    };
-    resolve_path(instr, "ArrayErase", callback);
+    resolve_path(
+        instr, "ArrayErase",
+        util::overload{[&](LstBase& list, size_t index) {
+                           if (index >= instr.prior_size) {
+                               bad_transaction_log("ArrayErase: Invalid index (index = %1, prior_size = %2)", index,
+                                                   instr.prior_size);
+                           }
+                           if (index >= list.size()) {
+                               bad_transaction_log("ArrayErase: Index out of bounds (%1 >= %2)", index, list.size());
+                           }
+                           if (instr.prior_size != list.size()) {
+                               bad_transaction_log("ArrayErase: Invalid prior_size (list size = %1, prior_size = %2)",
+                                                   list.size(), instr.prior_size);
+                           }
+
+                           list.remove(index, index + 1);
+                       },
+                       [&](LstBase&) {
+                           bad_transaction_log("Invalid path for ArrayErase (list)");
+                       },
+                       [&](SetBase&) {
+                           bad_transaction_log("Invalid path for ArrayErase (set)");
+                       },
+                       [&](Dictionary&) {
+                           bad_transaction_log("Invalid path for ArrayErase (dictionary)");
+                       },
+                       [&](Dictionary&, Mixed) {
+                           bad_transaction_log("Invalid path for ArrayErase (dictionary, key)");
+                       },
+                       [&](Obj&, ColKey) {
+                           bad_transaction_log("Invalid path for ArrayErase (obj, col)");
+                       }});
 }
 
 void InstructionApplier::operator()(const Instruction::Clear& instr)
 {
-    auto callback = util::overload{
-        [&](LstBase& list) {
-            list.clear();
-        },
-        [](Dictionary& dict) {
-            dict.clear();
-        },
-        [](SetBase& set) {
-            set.clear();
-        },
-        [&](auto&&...) {
-            bad_transaction_log("Invalid path for Clear");
-        },
-    };
-
-    resolve_path(instr, "Clear", callback);
+    resolve_path(instr, "Clear",
```

Comment:
Same as above

Reply:
This was the only case where VS 2019 actually didn't work with std::move, and I've added a comment about it and changed the rest to be consistent.